resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/delegates/MediaTypeHeaderDelegate.java

```diff
@@ -7,17 +7,35 @@
 import javax.ws.rs.ext.RuntimeDelegate;
 
 import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
 public class MediaTypeHeaderDelegate implements RuntimeDelegate.HeaderDelegate
 {
+   private static final int MAX_MT_CACHE_SIZE = Integer.getInteger("org.jboss.resteasy.plugins.delegates.max_mediatype_cache_size", 200);
+   private static final Map<String, MediaType> map = MAX_MT_CACHE_SIZE > 0 ? new ConcurrentHashMap<String, MediaType>() : null;
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#ConcurrentHashMap(int)

initial size could be set to the limit